packages/opentelemetry-tracing/test/common/export/BatchSpanProcessorBase.test.ts

```diff
@@ -141,6 +151,20 @@ describe('BatchSpanProcessorBase', () => {
       assert.strictEqual(exporter.getFinishedSpans().length, 0);
     });
 
+    it('should not export unsampled spans', async () => {
+      const processor = new BatchSpanProcessor(exporter, defaultBufferConfig);
+      const spy: sinon.SinonSpy = sinon.spy(exporter, 'export') as any;
+
+      const span = createUnsampledSpan(`${name}_0`);
```

Comment:
Actually this doesn't test the new code path as `AlwaysOffSampler` returns `NOT_RECORD` and as a result `onStart()` is not called at all.

Reply:
I'll modify the `createUnsampledSpan` helper to use the `RECORD_ONLY` decision, but in this case I don't think it makes a difference. The purpose of the helper is purely to generate a valid span, and doesn't actually add the span processor. Instead, the convention in this test suite is to produce a span and manually call the processor methods, which is what the new test follows. Most of the tests omit the call to `onStart` since in `BatchSpanProcessorBase`, it's a no-op.

Reply:
But your testcase is not calling `processor.onStart(span)`. If you just add this call it's also fine.

Reply:
Added the missing calls to the new test and all existing tests in the 2 relevant suites.